### PR TITLE
Support deactivation when a downstairs is offline

### PR DESCRIPTION
**Pull request by leftwo**
This adds support for deactivation of the upstairs when one or more downstairs is offline.

### CODE REVIEW

File: upstairs/src/downstairs.rs [hunk 1]
```diff
@@ -812,7 +812,7 @@ impl Downstairs {
          * TODO: Handle deactivation when a downstairs is offline.
```
[the remark this answers is not in the document]
Comment (leftwo) [reply upstairs/src/downstairs.rs:812]: Yes, and actually,I wonder if this check for Offline should even happen, as we don't 
do anything different here if there are Offline downstairs other than print a message.

I think I'll take the comment out of this method and move it where it makes better sense.

**Review by mkeeter — APPROVED** [review]

File: upstairs/src/client.rs [hunk 2]
```diff
@@ -1082,7 +1082,14 @@ impl DownstairsClient {
                     | DsState::Replay
                     | DsState::LiveRepair
                     | DsState::LiveRepairReady
+                    | DsState::Offline
```
Comment (mkeeter) [inline upstairs/src/client.rs:1085]: Is this transition actually a thing?  It looks like we added an `Offline` → `Faulted` transition, not directly `Offline` → `Deactivated`

Comment (leftwo) [reply upstairs/src/client.rs:1085]: Yeah, if we are offline and there are no jobs in the work queue, we will go from `Offline` to `Deactivated`

**Review by jmpesp — APPROVED** [review]

File: upstairs/src/client.rs (hunk 2 above)
Comment (mkeeter) [reply upstairs/src/client.rs:1085]: Got it, I see that transition happens before the new `Offline` → `Faulted` transition

**Pull request merged**